lib/datadog/appsec/component.rb

```diff
@@ -8,7 +8,7 @@ module AppSec
     class Component
       class << self
         def build_appsec_component(settings)
-          return unless settings.enabled
+          return unless defined?(Datadog::AppSec::Configuration) && settings.appsec.enabled
```

Comment:
I initially suggested `defined?`, but looking at the code here, I think we could probably get away with `respond_to?(:appsec)`. 

That would probably make it way easier to test -- we could just pass in a mock settings object that did not have an appsec entry.

Reply:
That makes a lot of sense. I like the idea of being able to test that functionality as well. 

Thanks for the suggestion 😄